src/trackers/process-crawl.js

```diff
@@ -41,23 +36,32 @@ async function processSite(siteName) {
     bar.tick()
 }
 
-async function processCrawl(fileList) {
+async function processCrawl() {
+    const reader = sharedData.config.crawlerDataLoc === 'postgres'
+        ? new PostgresDataReader(sharedData.config.crawlId, sharedData.config.regionCode)
+        : new JSONFileDataReader(sharedData.config.crawlerDataLoc)
     console.time("runtime")
-    
-    for (let i=0; i < fileList.length; i += sharedData.config.parallelism) {
-        const sites = []
-        for (const site of fileList.slice(i, i + sharedData.config.parallelism)) {
-            sites.push(processSite(site))
+    try {
+        bar = new Progress('Process crawl [:bar] :percent', {width: 40, total: await reader.length()})
+
+        let sites = []
+        for await (const siteData of reader.iterator()) {
+            if (sites.length >= sharedData.config.parallelism) {
+                // wait for batch to finish before reading more site data
+                await Promise.allSettled(sites)
+                sites = []
+            }
+            sites.push(processSite(siteData))
         }
         await Promise.allSettled(sites)
+        crawl.finalizeRequests()
+        crawl.writeSummaries()
+        console.log(`${chalk.blue(crawl.stats.sites)} sites processed\n${chalk.blue(crawl.stats.requests)} requests processed\n${chalk.blue(crawl.stats.requestsSkipped)} requests skipped`)
+    } finally {
```

Comment:
if there was an error I'd print it out - we want to know that the process haven't finished successfully

Reply:
Omitting the catch block will let the exception bubble up to the parent scope, so it will still be reported if not caught somewhere else.